test/simulator/metrics-collector-simulator.py

```diff
@@ -0,0 +1,355 @@
+                    return err
+
+            try:
+                deploymentAPI = hubClient.resources.get(api_version="apps/v1",
+                                                        kind="Deployment")
+
+                deploymentAPI.delete(name="metrics-collector-deployment",
+                                     namespace=clusterName)
+            except ApiException as err:
+                if err.status != 404:
+                    return err
+
+            try:
+                serviceAccountAPI = hubClient.resources.get(
+                    api_version="v1", kind="ServiceAccount")
+
+                serviceAccountAPI.delete(
+                    name="endpoint-observability-operator-sa",
+                    namespace=clusterName)
+            except ApiException as err:
+                if err.status != 404:
+                    return err
+
+            try:
+                serviceAccountAPI = hubClient.resources.get(
+                    api_version="v1", kind="ServiceAccount")
+
+                serviceAccountAPI.delete(
+                    name="endpoint-observability-operator-sa",
+                    namespace=clusterName)
+            except ApiException as err:
+                if err.status != 404:
+                    return err
+
+            secretAPI = hubClient.resources.get(api_version="v1",
+                                                kind="Secret")
+            try:
+                secretAPI.delete(
+                    name=
+                    "observability-controller-open-cluster-management.io-observability-signer-client-cert",
+                    namespace=clusterName)
+            except ApiException as err:
+                if err.status != 404:
+                    return err
+
+            try:
+                secretAPI.delete(name="observability-managed-cluster-certs",
+                                 namespace=clusterName)
+            except ApiException as err:
+                if err.status != 404:
+                    return err
+
+            try:
+                configMapAPI = hubClient.resources.get(api_version="v1",
+                                                       kind="ConfigMap")
+                configMapAPI.delete(
+                    name="metrics-collector-serving-certs-ca-bundle",
+                    namespace=clusterName)
+            except ApiException as err:
+                if err.status != 404:
+                    return err
+
+    return None
+
+
+def watchManagedClusters(hubClient: dynamic.DynamicClient,
+                         clusterNamePrefix: str) -> None:
+    groupVersion = "cluster.open-cluster-management.io/v1"
+
+    # it seems the kind string is case sensitive
+    kind = "ManagedCluster"
+
+    api = hubClient.resources.get(api_version=groupVersion, kind=kind)
+
+    print("start watch: {}/{}".format(groupVersion, kind))
+
+    set = {}
+
+    for e in api.watch():
+        print(e['type'])
+
+        if e['type'] == "ADDED":
```

Comment:
do we need to watch "DELETED" event?

Reply:
I was thinking for using the `--clean` flag to delete the simulators, so I didn't watch the `DELETED`